assets/js/omniSelectEngine.js

```diff
@@ -0,0 +1,394 @@
+        .length;
+      var hasSelection =
+        $(".select2.select2-container").attr("data-visible-selected") &&
+        $(".select2.select2-container").attr("data-visible-selected").length >
+          0 &&
+        searchFieldInputLength == 0;
+      switch (e.keyCode) {
+        case 8:
+          if (hasSelection) {
+            destroyOmniSearch();
+            initOmniSearch();
+          }
+
+          break;
+        case 27:
+          destroyOmniSearch();
+          break;
+      }
+    });
+  }
+
+  // Api
+  function getAllProjects() {
+    return callApi("leantime.rpc.projects.getAll", {});
+  }
+  function getUserTickets() {
+    return callApi("leantime.rpc.tickets.getAll", {
+      searchCriteria: {
+        userId: userId,
+      },
+    });
+  }
+  function callApi(method, params) {
+    return new Promise((resolve, reject) => {
+      $.ajax({
+        url: leantime.appUrl + "/api/jsonrpc/",
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        data: JSON.stringify({
+          method: method,
+          jsonrpc: "2.0",
+          id: "1",
+          params: params,
+        }),
+        success: resolve,
+        error: reject,
+      });
+    });
+  }
+
+  // Get data for initial load.
+  function getOmnisearchData() {
+    var availableTags = [];
+
+    let projects = getAllProjects().then((data) => {
+      var projects = data.result;
+      const projectGroup = {
+        id: "project",
+        text: "Projects",
+        children: [],
+        index: 1,
+      };
+      projects.forEach((project) => {
+        let option = {
+          id: project.id,
+          text: project.name,
+          type: "project",
+          client: project.clientName,
+        };
+        projectGroup.children.push(option);
+      });
+      availableTags.push(projectGroup);
+    });
+
+    let tickets = getUserTickets().then((data) => {
+      var result = data.result;
+      let tickets = result.filter((result) => result.type === "task");
+      const ticketGroup = {
+        id: "task",
+        text: "To-Do´s",
+        children: [],
+        index: 2,
+      };
+      tickets.forEach((ticket) => {
+        let option = {
+          id: ticket.id,
+          text: ticket.headline,
+          type: ticket.type,
+          tags: ticket.tags,
+          sprintName: ticket.sprintName,
+          projectId: ticket.projectId,
+          projectName: ticket.projectName,
+        };
+        ticketGroup.children.push(option);
+      });
+      availableTags.push(ticketGroup);
+    });
+
+    // Sort data by index and return to select2.
+    projects.then(() => {
+      tickets.then(() => {
+        availableTags.sort(function (a, b) {
+          return a.index - b.index;
+        });
+        reinitOmniSearchWithData(availableTags);
+        omniSelectElement.removeClass("loading");
+      });
+    });
+  }
+
+  function fuzzySearch(needle, haystack) {
```

Comment:
Fuzzy search are normally used to `searches for text that matches a term closely instead of exactly`, but you are more doing a sub-part string match search...

Could an simple https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes be used?

Reply:
My idea with this is to enable kind of a sub-part string search.
Say you have a task called "genvej rulles i produktion"
You search for "genvej prod" (because that's what you remember its approximately called).
.includes would return nothing where this would return the task.